packages/ra-core/src/controller/CreateController.js

```diff
@@ -48,14 +49,6 @@ import { crudCreate as crudCreateAction } from '../actions';
  *     export default App;
  */
 class CreateController extends Component {
-    getBasePath() {
-        const { location } = this.props;
-        return location.pathname
```

Comment:
I don't understand, removing this will probably cause regressions

Reply:
Why ?

Reply:
The e2e tests say it's ok ;

Reply:
because we take into account only the final part of the URL. An URL looking like `/admin/posts` won't work anymore.

Reply:
I'll test that but at any rate, this should not be the responsability of those components. They don't need to be aware of routing. The logic you mention could be applied in the `Resource` component

Reply:
agreed

Reply:
> because we take into account only the final part of the URL.

What made you think that ?

In `Resource` at [L75](https://github.com/marmelab/react-admin/pull/1942/files#diff-3332c15e881ebff85a83708a32c91a1aR75):

```js
const basePath = match.url;
```

`match.url` will be whatever the url is at this point: `/posts` or `/admin/posts` or am I missing something ?

Reply:
Btw, would it be better to extract those changes in a dedicated PR ?